transaction/std/src/transaction_builder.rs

```diff
@@ -108,12 +111,16 @@ impl TransactionBuilder {
         Ok((tx_out, confirmation))
     }
 
-    /// Sets the tombstone block.
+    /// Sets the tombstone block, clamping to smallest pubkey expiry value.
     ///
     /// # Arguments
     /// * `tombstone_block` - Tombstone block number.
-    pub fn set_tombstone_block(&mut self, tombstone_block: u64) {
-        self.tombstone_block = tombstone_block;
+    pub fn set_tombstone_block(&mut self, tombstone_block: u64) -> u64 {
+        self.tombstone_block = core::cmp::min(
+            tombstone_block,
+            self.fog_pubkeys.smallest_pubkey_expiry_value(),
```

Comment:
I think it would be better to only enforce the expiries of the fog pubkeys that were used to construct fog hints, rather than enforcing all fog pubkeys contained in `FullyValidatedFogPubkeys`.

If `FullyValidatedFogPubkeys` contains more fog pubkeys than are actually used, then this technique runs the risk of unnecessarily restricting the `tombstone_block` to a lower value than it would otherwise need to.

Reply:
yeah, okay, we can keep a vec of actually used expiry values in the transaction builder or something

Reply:
or just compute the min as we go.

in an early draft of this PR i made `create_output` a private member funciton of TxBuilder but then a bunch of unit tests would have had to change. but i can work through that.